src/object.di

```diff
@@ -595,6 +595,111 @@ inout(V) get(K, V)(inout(V[K])* aa, K key, lazy inout(V) defaultValue)
     return (*aa).get(key, defaultValue);
 }
 
+private template object_anySatisfy(alias F, T...)
+{
+    static if (T.length == 0)
+    {
+        enum object_anySatisfy = false;
+    }
+    else static if (T.length == 1)
+    {
+        enum object_anySatisfy = F!(T[0]);
+    }
+    else
+    {
+        enum object_anySatisfy =
+            object_anySatisfy!(F, T[ 0  .. $/2]) ||
+            object_anySatisfy!(F, T[$/2 ..  $ ]);
+    }
+}
+
+// Somehow fails for non-static nested structs without support for aliases
+private template object_hasElaborateDestructor(T...)
+{
+    static if (is(T[0]))
+        alias S = T[0];
+    else
+        alias S = typeof(T[0]);
+
+    static if (is(S : E[n], E, size_t n) && S.length)
+    {
+        enum bool object_hasElaborateDestructor = object_hasElaborateDestructor!E;
+    }
+    else static if (is(S == struct))
+    {
+        enum object_hasElaborateDestructor = __traits(hasMember, S, "__dtor")
+            || object_anySatisfy!(.object_hasElaborateDestructor, S.tupleof);
+    }
+    else
+        enum bool object_hasElaborateDestructor = false;
+}
+
+private void destructRecurse(S)(ref S s)
+    if (is(S == struct))
+{
+    static if (__traits(hasMember, S, "__dtor"))
+        s.__dtor();
+
+    foreach_reverse (i, ref field; s.tupleof)
+    {
+        static if (object_hasElaborateDestructor!(typeof(field)))
+            destructRecurse(field);
+    }
+}
+
+private void destructRecurse(E, size_t n)(ref E[n] arr)
+{
+    static if (object_hasElaborateDestructor!E)
+    {
+        foreach_reverse (ref elem; arr)
+            destructRecurse(elem);
+    }
+}
+
+// Somehow fails for non-static nested structs without support for aliases
+private template object_hasElaborateCopyConstructor(T...)
+{
+    static if (is(T[0]))
+        alias S = T[0];
+    else
+        alias S = typeof(T[0]);
+
+    static if (is(S : E[n], E, size_t n) && S.length)
+    {
+        enum bool object_hasElaborateCopyConstructor = object_hasElaborateCopyConstructor!E;
+    }
+    else static if (is(S == struct))
+    {
+        enum object_hasElaborateCopyConstructor = __traits(hasMember, S, "__postblit")
+            || object_anySatisfy!(.object_hasElaborateCopyConstructor, S.tupleof);
+    }
+    else
+        enum bool object_hasElaborateCopyConstructor = false;
+}
+
+// Public and explicitly undocumented
+void postblitRecurse(S)(ref S s)
```

Comment:
It is private and the delegate/function pointer it returns does not have properly propagated attributes; it is the same problem `destroy` had by going through `TypeInfo`.